### DISCUSSION TITLE
What is this magic?

### DISCUSSION THREAD

**Discussion by robotlightsyou** [Slow-Scallion4183/Adventure-Colussus#52] [Q&A] [answered]
character = entities.Human(**(dict(character) | character_dict))

I've been thinking about this line for almost a week and I just don't get it. I understand "|" to be the 'bitwise or', but I don't really understand how that is different from regular or. I've used this as the union operator for sets, is this somehow saying "pick the non-falsey value from the options"? I've also tried to recreate this with 2 dictionaries, or trying to call dict() on a test class, but I keep getting errors in the interpreter.

If someone could either explain this a little, or point me to the name of what functionality is being leveraged here I would greatly appreciate it, my google-fu is failing me.

**Comment by wireboy5** [comment, marked as answer]
It is the union operator for dictionaries.

**Reply by robotlightsyou** [reply]
Much obliged.
